SCons/Tool/MSCommon/README.rst

```diff
@@ -0,0 +1,125 @@
+README - SCons.Tool.MSCommon 
+############################
+
+Compiler Detection Logic
+========================
+
+**WARNING: the compiler detection logic documentation below is likely out-of-date.**
+
+The verbatim content below is taken in its entirety from the existing ``MSCommom/README``
+file.  In the future, the the compiler detection logic documentation will will be updated
+and converted to the current document format.
+
+::
+
+    This is the flow of the compiler detection logic:
+
+    External to MSCommon:
+
+      The Tool init modules, in their exists() routines, call -> msvc_exists(env)
+
+    At the moment, those modules are:
+      SCons/Tool/midl.py
+      SCons/Tool/mslib.py
+      SCons/Tool/mslink.py
+      SCons/Tool/msvc.py
+      SCons/Tool/msvs.py
+
+    env may contain a version request in MSVC_VERSION, but this is not used
+    in the detection that follows from msvc_exists(), only in the later
+    batch that starts with a call to msvc_setup_env().
+
+    Internal to MSCommon/vc.py:
+
+    + MSCommon/vc.py:msvc_exists:
+    | vcs = cached_get_installed_vcs(env)
+    | returns True if vcs > 0
+    |
+    +-> MSCommon/vc.py:cached_get_installed_vcs:
+      | checks global if we've run previously, if so return it
+      | populate the global from -> get_installed_vcs(env)
+      |
+      +-> MSCommon/vc.py:get_installed_vcs:
+        | loop through "known" versions of msvc, granularity is maj.min
+        |   check for product dir -> find_vc_pdir(env, ver)
+        |
+        +-> MSCommon/vc.py:find_vc_pdir:
+          | From the msvc-version to pdir mapping dict, get reg key base and value
+          | If value is none -> find_vc_pdir_vswhere(ver, env)
+          |
+          +-> MSCommon/vc.py:find_vc_pdir_vswhere:
+            | From the vc-version to VS-version mapping table get string
+            | Figure out where vswhere is -> msvc_find_vswhere()
+            | Use subprocess to call vswhere, return first line of match
+            /
+          | else get product directory from registry (<= 14.0)
+          /
+        | if we found one -> _check_cl_exists_in_vc_dir(env, pdir, ver)
+        |
+        +-> MSCommon/vc.py:_check_cl_exists_in_vc_dir:
+          | Figure out host/target pair
+          | if version > 14.0 get specific version by looking in
+          |    pdir + Auxiliary/Build/Microsoft/VCToolsVersion/default.txt
```

Comment:
> __[Secret Scan]__ It may be dangerous to commit the AWS secret access key. ([view](https://sider.review/gh/repos/104670160/pulls/4174#issue-7331632))
>
> <details>
>
> AWS secret access key is an important credential that allows you to use AWS programmatically.
> Once this leaks, an attacker can access and operate your AWS resources.
>
> See also:
> * https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html
>
> | Rule | Severity |
> | ---- | -------- |
> | `sider.secrets.aws.secret_access_key` | `warning` |
>
> You can [close this issue](https://sider.review/gh/repos/104670160/pulls/4174#issue-7331632) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

Sider got fooled, it thinks this might contain an AWS secret key.

The line it's flagging sure doesn't look like that to me:
```txt
          |    pdir + Auxiliary/Build/Microsoft/VCToolsVersion/default.txt
```

Reply:
closed that one.. Weird.